process/block/baseProcess.go

```diff
@@ -556,7 +556,8 @@ func (bp *baseProcessor) requestHeadersIfMissing(
 		}
 
 		requested++
-		go bp.requestHeaderByShardAndNonce(shardId, nonce)
+		// TODO : change the epoch parameter to be the right one
```

Comment:
at this moment in case of epoch changes you do not know which nonce in which epoch is. There are several edge cases. asking with the lastNotarizedHdr.Epoch is ok.

Reply:
not relevant anymore